### ISSUE TITLE
Licence for the data submitted

### ISSUE THREAD

**Issue by arthurlutz** [MrBreadWater/project-kronos-data-collector#4]
It would be useful to provide information about the licence the data that is submitted is subjected to. Does a contributing user keep his copyright on it ? Can the data be contributed as Creative Commons so that other people can build neural networks on top of that dataset ? Does one have access to the rest of the data if interrested in doing research ? Could the ODBL https://opendatacommons.org/licenses/odbl/ be used ?

**Comment by MrBreadWater** [comment]
By using the plugin, you agree to allow Project Kronos to privately store the timelapses and snapshots as needed, and use it for the purposes of the project. What you do with _your_ timelapses (stored locally on your machine) is up to you, feel free to make your own dataset. But the dataset itself likely won't be public, due mainly to complications with copyright, possible privacy, etc. By possible privacy, I mean to say that posting everyone's timelapses publically does seem somewhat wrong. That was not the agreement that they had when they downloaded the timelapse. Even though there's no personally identifiable information there except for anything a user has their camera pointed at that could be used to identify them, it is likely against many users' wishes.

I won't close this one, as this may have some legitimate discussion.

**Comment by arthurlutz** [comment]
@MrBreadWater thanks for considering that this is a legitimate discussion. 

I think this issue can be considered from the point of view of "adoption of the project". I think a lot of 3d printer users are aware of the issues around the "commons" (cf. the patent-free technology and the rise of open hardware printers), the ecosystem is widely open source, the models are widely creative commons... 

To draw a parallel, I'm happy to contribute GPS tracking data to OSM so that it improves the map which is a common good, but less likely to contribute that same data to google or waze (which is google too).

**Comment by guysoft** [comment]
How about opt-in to share your photos publicly, even for individual prints?

The reason is that if there is a public dataset it would be easier to improve its results. Currently the model means only a handful of humans can actually work and train the network.

**Comment by tedder** [comment]
I was coming here to say that, guysoft. As it stands this is a black box to everyone but mrbreadwater. It would be nice to extend it, and perhaps also have a "share metadata" box to make training easier- that would let it signal a "cancel", for instance.

Having the images public lets humans help with training- "this is a failed print", for instance.

**Comment by arossbach10** [comment]
I think the labels would even be available via Octoprint. OctoPrint stats plugin or something.

**Comment by arthurlutz** [comment]
For people interested in crowdsourcing such a dataset (if it existed) I would recommend taking a look at https://pybossa.com/ (for concrete examples try out a task at https://crowdcrafting.org/ )

**Comment by MrBreadWater** [comment]
I've been mulling over this for some time now, and I've decided that I'm going to create this project as a private product. It will remain a black box service, but it will have some sort of a free tier that allows you to use the plugin normally, albeit with a few less bells and whistles than the paid service. 

I know that releasing this entirely open source would be helpful for the community and creation, but I started on this project in the first place in hopes of using it to start a career.

So I've come up with this: People who contribute using the Project Kronos Data Collector will get the opportunity to add a username or email to a database. _(This will be kept separate from the data, no identifying information will be attached to your data, all data is anonymous. Just have to keep this perfectly clear for legal purposes.)_ Later on, this database will give users the opportunity to receive some sort of compensation.

I will also add a clause to the Project Kronos Data Collector agreement that allows for the possibility of releasing the dataset later on for public usage. That way, at the end of the program's lifespan, I could release the dataset publicly.

Finally, I would like to hear your opinions on what might be a fair compensation for contributing to the dataset. I was thinking some free usage of the premium tier? But I'd like to hear what you think.
